lumicks/pylake/channel.py

```diff
@@ -504,6 +504,172 @@ def plot(self, start=None, **kwargs):
+                    f'"vertical", got {annotation_direction}.'
+                )
+
+    def highlight_time_range(
+        self,
+        item,
+        annotation=None,
+        annotation_direction=None,
+        color=None,
+        label=None,
+        start=None,
+        alpha=0.5,
+        **kwargs,
+    ):
+        """Highlight some points or regions in a plot.
+
+        Parameters
+        ----------
+        item : Marker | tuple(int, int) | int | str
+            An item with a start and stop property such as a `Marker` or `CalibrationItem`, a
+            tuple of timestamps, a single timestamp or a string in time format (e.g. "5s" for
+            5 seconds).
+        annotation : str | None
+            String to annotate with
+        annotation_direction : "horizontal" | "vertical" | None
+            Which direction to show the text when annotating.
+        color : matplotlib.color | None
+            Matplotlib color
+        label : str | None
+            Label to use in legend entry if a legend is plotted
+        start : int | None
+            Plot with respect to a different "zero" timestamp.
+        alpha : float
+            Opacity (0 is transparent, 1.0 is opaque). Default: 0.5.
+        **kwargs :
+            Arguments forwarded to `matplotlib.axvline` and `matplotlib.axvspan`.
+
+        Examples
+        --------
+        ::
+
+            from lumicks import pylake
+
+            f = lk.File("data.h5")
+
+            plt.figure()
+            slc = ["Distance"]["Distance 1"]
+            slc.plot()
+
+            # Draws the time region spanned by marker 1
+            slc.highlight_time_range(f.markers["1"])
+
+            # Draws an annotation at the 1-second mark
+            slc.highlight_time_range("1s", annotation="first second")
+
+            # Draws a highlighted region from 1 to 3 seconds and marks it with the text "region"
+            slc.highlight_time_range(("1s", "3s"), annotation="region")
+
+            # Draws a highlighted region from 4 to 5 seconds, marks it, but forces text vertically
+            slc.highlight_time_range(("1s", "3s"), annotation="region", annotation_direction="vertical")
+
+
+            plt.figure()
+            f.force1x.plot()
+
+            item = f.force1x.calibration[1]
+
+            # Fills the region between calibration_item.start and calibration_item.stop
+            f.force1x.highlight_time_range(item, annotation="calibration 1")
+
+            # Draws a line with annotation at the time the calibration was applied.
+            f.force1x.highlight_time_range(item.applied_at, annotation="calibration 1 applied")
+        """
+        import matplotlib.pyplot as plt
+
+        def to_seconds(timestamp):
+            start_ts = self.start if start is None else start
+
+            if isinstance(timestamp, str):
+                timestamp = to_timestamp(timestamp, start_ts, self.stop)
+
+            return (timestamp - start_ts) / 1e9
```

Comment:
I know it's not needed per se but considering we have the `to_timestamp` function in `detail.timeindex` could throw this one there too

Reply:
Fair point. I've split it out now and added a little test for it.